ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ExplainExecutor.java

```diff
@@ -118,20 +125,29 @@ private static QueryDescription explainStatement(
               new IllegalStateException("The provided statement did not run a ksql query"));
     }
 
-    return QueryDescriptionFactory.forQueryMetadata(metadata);
+    return QueryDescriptionFactory.forQueryMetadata(
+        metadata,
+        Collections.singletonMap(
+            new KsqlHostInfoEntity(sessionProperties.getKsqlHostInfo()),
+            metadata.getState()));
   }
 
   private static QueryDescription explainQuery(
       final String queryId,
-      final KsqlExecutionContext executionContext
+      final KsqlExecutionContext executionContext,
+      final SessionProperties sessionProperties
   ) {
     final PersistentQueryMetadata metadata = executionContext
         .getPersistentQuery(new QueryId(queryId))
         .orElseThrow(() -> new KsqlException(
             "Query with id:" + queryId + " does not exist, "
                 + "use SHOW QUERIES to view the full set of queries."));
 
-    return QueryDescriptionFactory.forQueryMetadata(metadata);
+    return QueryDescriptionFactory.forQueryMetadata(
+        metadata,
+        Collections.singletonMap(
+            new KsqlHostInfoEntity(sessionProperties.getKsqlHostInfo()),
+            metadata.getState()));
```

Comment:
I changed it to pass down the `KsqlHostInfoEntity` object, but the map would still need to be made separately in each function since the` QueryMetadata` isn't available for the Explain statement case.

Reply:
Actually, I realized, for `explainStatement`, the `ksqlQueryHostState` should actually be an emptyMap since we're not actually running the statement yet.